wordpress-mobile/WordPress-Android
# Issue/12028 fix block site action

Parent issue #12028

Merge instructions:
1. Make sure https://github.com/wordpress-mobile/WordPress-Android/pull/12791#issuecomment-679961254 is merged
2. Remove "Not Ready for Merge" label
3. Merge this PR

This PR refactors the BlockSite action into a useCase and re-uses that usecase in the ReaderPostListFragment and the "new" discover tab VM.

Since the useCase needs to return multiple results (in progress states), I decided to implement the use case using Kotlin Flows. The "Undo" action was causing a lot of design issues, since it's technically another action within a single usecase. I discussed the design with Vojta and he suggested to extract both SnackBarHolders (including the Undo action) and refresh posts events from the UseCase and keep it single focused - block site action. 
I decided to go with this solution after some consideration as it makes a lot of sense. The usecase is very focused now and it isn't tied to the view layer at all. The `ReaderPostCardActionsHandler` is responsible for transforming the usecase states into corresponding Ui Actions - show snackbar, refresh post.
Let me know what you think about it :). I'll add tests when we decide this is the solution we want to go with.

To test - Enable RI FF
1. Open Reader
2. Open Discover tab
3. Click on the more button on one of the items
4. Click on "Block Site"
5. Notice all posts from this site disappear
6. Notice Snackbar message with "Undo" action is shown
7. Click on "Undo" and notice the posts re-appear
--------------------
1. Open Reader
2. Open Discover tab
3. Click on the more button on one of the items
4. Turn on airplane mode
5. Click on "Block Site"
6. Notice an error snackbar appears

Perform the same steps on the Following Tab.

PR submission checklist:
